ts/Series/Pie/PieSeries.ts

```diff
@@ -274,11 +274,11 @@ class PieSeries extends Series {
 
     /**
      * Define hasData function for non-cartesian series. Returns true if the
-     * series has points at all.
+     * series has at least one visible point (#23235)
      * @private
      */
     public hasData(): boolean {
-        return !!this.dataTable.rowCount;
+        return this.points.some((point): boolean => point.visible !== false);
```

Comment:
Given that `point.visible` is [never undefined](https://github.com/highcharts/highcharts/blob/f090a057f64544895ed73ca32df1feb4f61f32e0/ts/Core/Series/Point.ts#L145), it should be sufficient with a boolean test:
```suggestion
        return this.points.some((point): boolean => point.visible);
```